lib/controllers/facebook-login.js

```diff
@@ -75,7 +75,11 @@ module.exports = function (req, res) {
             return oauth.errorResponder(req, res, err);
           }
 
-          var nextUrl = url.parse(req.query.next || '').path || (resp.created ? config.web.register.nextUri : config.web.login.nextUri);
+          var nextUrl = oauth.common.consumeRedirectUri(req, res);
+
+          if (!nextUrl) {
+            nextUrl = url.parse(req.query.next || '').path || (resp.created ? config.web.register.nextUri : config.web.login.nextUri);
```

Comment:
Do we need to keep this: ` url.parse(req.query.next || '').path` ? It seems that this will never be available during a social login flow, and that is why we're doing the cookie workaround?